backend/app/routes/admin/getAdminList.js

```diff
@@ -0,0 +1,34 @@
+const Admin = require('../../models/Admin');
+
+module.exports = async (req, res, next) => {
+  const options = req.query;
+  console.log(options);
+  const responseArr = await Admin.find({}).then(function (users) {
+    const response = [];
+    // users[0]["isSuperAdmin"] = false; uncomment this to check all conditions are working
+
+    // In database some admins didn't insert their names so it returned null. Hence list of emails is being returned as this is inserted by each admin
+    const adminPropertyToBeReturned = 'email';

```

Comment:
@atharva-upadhye-01  You need to use aggregate functions for this
This code is not scalable. Let me know if you need some inputs from me on aggregate functions
I will review your code thoroughly tomorrow.

Reply:
Hello @jackfrost13, I searched about aggregate functions and used it as below while trying to make code more scalable. Please give review on this.

```js
const Admin = require('../../models/Admin');

function getPropsFromAdminData(a) {
  let json = '{';
  if (!a.includes('_id')) json += '"_id":0,';
  a.forEach(e => json += `"${e}": "$${e}",`);
  json = json.replace(/.$/, '}')
  console.log(JSON.parse(json));
  return { $project: JSON.parse(json) }
}

module.exports = async (req, res, next) => {
  const options = req.query;
  const propsFromAdminData = ['email', 'firstName']
  let resArray;
  try {
    resArray = await Admin.aggregate([
      { $match: { isSuperAdmin: JSON.parse(options.isSuperAdmin) } },
      getPropsFromAdminData(propsFromAdminData)
    ])
  } catch (error) {
    console.error(`${error.message}, hence returning all.`);
    resArray = await Admin.aggregate([
      getPropsFromAdminData(propsFromAdminData)
    ])
  }
  res.status(200).send(resArray);
  return next();
};
```

Reply:
not exactly the correct way of doing this i believe. Ping me on slack once, let's discuss there, I can show you a few samples

Reply:
The pagination is also not implemented here

Reply:
Yes I need some guidance regarding pagination too.
I have pinged you on slack.